lib/oauther.ex

```diff
@@ -72,19 +72,25 @@ defmodule OAuther do
   end
 
   defp base_string(verb, url, params) do
-    [verb, URI.parse(url), params]
-    |> Stream.map(&normalize/1)
-    |> Enum.map_join("&", &percent_encode/1)
+    {base_url, q_params} = parse_url(url)
+    [String.upcase(verb), base_url, normalize_params(params, q_params)]
+      |> Enum.map_join("&", &percent_encode/1)
+  end
+
+  defp parse_url(url) do
+    uri = URI.parse(url)
+    {"#{uri.scheme}://#{uri.host}#{uri.path}", parse_query_params(uri) }
```

Comment:
I tried again, tho my elixir fu wasn't quite up to implementing `Map.get_and_update/3` as you suggest.